### DISCUSSION TITLE
Simple read/write while connected remotely via gateway

### DISCUSSION THREAD

**Discussion by jpomo** [ottowayi/pycomm3#256] [Q&A]
I am connected to an AB CompactLogix wirelessly using an Ewon VPN Gateway.

I confirmed the connection by pinging the PLC's static IP through the terminal.

Attempting to read or write to tags using pycomm3 gives a TimeoutError.

I know the address of both the Gateway and the PLC - is there a way for me to route pycomm's read/write requests 'through' the VPN?

Any thoughts are appreciated!

**Comment by dhunt84971** [comment]
I could be wrong, but I would think, if you can ping it you should be able to read from the PLC without any special routing on the pycomm side.  Maybe you need to open the EIP port 44818 through the Ewon VPN Gateway?

**Reply by ottowayi** [reply]
yeah that's what I'm thinking, since ping isn't TCP the ports don't matter.  @jpomo can you share a log?  That will help identify the issue, whether it's timing out on opening the socket or if it's a CIP unconnected send timeout (which is unlikely, since that's ~5s).
